pkg/server/server_sql.go

```diff
@@ -1792,6 +1814,34 @@ func (s *SQLServer) preStart(
 	return nil
 }
 
+// updateLocalityAddressesForSharedSecondaryTenants updates the locality addresses of this sql instance
+// if the instance is a shared secondary tenant. We update the port part of each address to that of the
+// advertised address as this will be stored in the sql_instances table and will be consulted by the
+// sqlInstanceDialer when attempting to dial this sql instance.
```

Comment:
I think what I want to know from this comment is why locality addresses aren't correct by default.

Reply:
In short, have removed this function now as its not needed.